core/src/main/java/org/elasticsearch/plugins/PluginInfo.java

```diff
@@ -57,15 +66,17 @@
     /**
      * Construct plugin info.
      *
+     * @param uberPlugin          the name of the uber plugin or null if this plugin is a standalone plugin
      * @param name                the name of the plugin
      * @param description         a description of the plugin
      * @param version             the version of Elasticsearch the plugin is built for
      * @param classname           the entry point to the plugin
      * @param hasNativeController whether or not the plugin has a native controller
      * @param requiresKeystore    whether or not the plugin requires the elasticsearch keystore to be created
      */
-    public PluginInfo(String name, String description, String version, String classname,
+    public PluginInfo(@Nullable String uberPlugin, String name, String description, String version, String classname,
                       boolean hasNativeController, boolean requiresKeystore) {
+        this.uberPlugin = uberPlugin;
```

Comment:
What is this link back to the meta plugin needed for?

Reply:
To keep track of the meta plugin name when the node loads the plugin but it's not needed. I removed it from the pr.